relay-metrics/src/aggregation.rs

```diff
@@ -1357,6 +1324,160 @@ impl<T: Iterator<Item = Bucket>> Iterator for CappedBucketIter<T> {
 
 impl<T: Iterator<Item = Bucket>> FusedIterator for CappedBucketIter<T> {}
 
+/// Check whether the aggregator has not (yet) exceeded its total limits. Used for health checks.
+#[derive(Debug)]
+pub struct AcceptsMetrics;
+
+/// Used only for testing the `AggregatorService`.
+#[cfg(test)]
+#[derive(Debug)]
+pub struct BucketCountInquiry;
+
+/// A message containing a vector of buckets to be flushed.
+///
+/// Handlers must respond to this message with a `Result`:
+/// - If flushing has succeeded or the buckets should be dropped for any reason, respond with `Ok`.
+/// - If flushing fails and should be retried at a later time, respond with `Err` containing the
+///   failed buckets. They will be merged back into the aggregator and flushed at a later time.
+#[derive(Clone, Debug)]
+pub struct FlushBuckets {
+    /// The project key.
+    pub project_key: ProjectKey,
+    /// The logical partition to send this batch to.
+    pub partition_key: Option<u64>,
+    /// The buckets to be flushed.
+    pub buckets: Vec<Bucket>,
+}
```

Comment:
It's a good idea to reorder these declarations, but it also makes the diff hard to read. What if we keep them where they were, and do the reordering in a separate PR that does not change anything functionally?

Reply:
If this is difficult to parse I can move those structs back.  I decided to move all in one place so it either to see how the API of this service looks like and one can see all the messages and `impl`s for them in one place.